Remote Web Inspector: [Cocoa] `inspectable` API

patrickangle opened the pull request:
#### a0a26311c18df6a6b7aa596ee06ee0b4257f765a
<pre>
Remote Web Inspector: [Cocoa] `inspectable` API
<a href="https://bugs.webkit.org/show_bug.cgi?id=245986">https://bugs.webkit.org/show_bug.cgi?id=245986</a>
rdar://100736163

Reviewed by Devin Rousso.

Add API to allow applications to explicitly mark a WKWebView or JSContext as inspectable. This flips assumptions about
inspectability of content. Previously the default state for inspectability was `true`, with a platform-specific deamon
making decisions about which applications would actually be inspectable on a system. This patch moves that decision into
WebKit/WebCore/JavaScriptCore instead, and flips the default for inspectability to be `false` unless an application
would have previously been always-inspectable. Those cases include:
- Applications with the `get-task-allow`/`com.apple.security.get-task-allow` development entitlements when linked
against an SDK without the API.
- Applications with the `com.apple.webinspector.allow`/`com.apple.private.webinspector.allow-remote-inspection`
entitlements.

These entitlements will no longer be necessary to enable inspection, and instead the new `inspectable` API should be
used.

* Source/JavaScriptCore/API/JSContext.h:
* Source/JavaScriptCore/API/JSContext.mm:
(-[JSContext inspectable]):
(-[JSContext setInspectable:]):
(-[JSContext _remoteInspectionEnabled]):
(-[JSContext _setRemoteInspectionEnabled:]):
* Source/JavaScriptCore/API/JSContextPrivate.h:
* Source/JavaScriptCore/API/JSContextRef.cpp:
(JSGlobalContextCreateInGroup):
(JSGlobalContextIsInspectable):
(JSGlobalContextSetInspectable):
(JSGlobalContextGetRemoteInspectionEnabled):
(JSGlobalContextSetRemoteInspectionEnabled):
* Source/JavaScriptCore/API/JSContextRef.h:
* Source/JavaScriptCore/API/JSContextRefPrivate.h:
* Source/JavaScriptCore/API/JSRemoteInspector.cpp:
(mainProcessHasEntitlement):
(defaultStateForRemoteInspectionEnabledByDefault):
(JSRemoteInspectorGetInspectionEnabledByDefault):
* Source/JavaScriptCore/API/JSRemoteInspector.h:
* Source/JavaScriptCore/API/tests/testIncludes.m:
* Source/JavaScriptCore/config.h:
* Source/JavaScriptCore/inspector/JSGlobalObjectInspectorController.cpp:
(Inspector::JSGlobalObjectInspectorController::developerExtrasEnabled const):
* Source/JavaScriptCore/inspector/remote/RemoteInspectionTarget.cpp:
(Inspector::RemoteInspectionTarget::remoteControlAllowed const):
(Inspector::RemoteInspectionTarget::inspectable const):
(Inspector::RemoteInspectionTarget::setInspectable):
(Inspector::RemoteInspectionTarget::pauseWaitingForAutomaticInspection):
(Inspector::RemoteInspectionTarget::setRemoteDebuggingAllowed): Deleted.
* Source/JavaScriptCore/inspector/remote/RemoteInspectionTarget.h:
* Source/JavaScriptCore/inspector/remote/RemoteInspector.h:
* Source/JavaScriptCore/inspector/remote/cocoa/RemoteInspectorCocoa.mm:
(Inspector::RemoteInspector::parentProcessAuditToken):
(Inspector::RemoteInspector::listingForInspectionTarget const):
* Source/JavaScriptCore/inspector/remote/glib/RemoteInspectorGlib.cpp:
(Inspector::RemoteInspector::listingForInspectionTarget const):
* Source/JavaScriptCore/inspector/remote/socket/RemoteInspectorSocket.cpp:
(Inspector::RemoteInspector::listingForInspectionTarget const):
* Source/JavaScriptCore/jsc.cpp:
(CommandLine::parseArguments):
(runJSC):
* Source/JavaScriptCore/runtime/JSGlobalObject.cpp:
(JSC::JSGlobalObject::setInspectable):
(JSC::JSGlobalObject::inspectable const):
(JSC::JSGlobalObject::setRemoteDebuggingEnabled): Deleted.
(JSC::JSGlobalObject::remoteDebuggingEnabled const): Deleted.
* Source/JavaScriptCore/runtime/JSGlobalObject.h:
* Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.h:
* Source/WebCore/config.h:
* Source/WebCore/page/Page.cpp:
(WebCore::Page::inspectable const):
(WebCore::Page::setInspectable):
(WebCore::Page::remoteInspectionAllowed const): Deleted.
(WebCore::Page::setRemoteInspectionAllowed): Deleted.
* Source/WebCore/page/Page.h:
* Source/WebCore/workers/service/context/ServiceWorkerThreadProxy.cpp:
(WebCore::ServiceWorkerThreadProxy::ServiceWorkerThreadProxy):
* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(WKPageGetAllowsRemoteInspection):
(WKPageSetAllowsRemoteInspection):
* Source/WebKit/UIProcess/API/Cocoa/APISerializedScriptValueCocoa.mm:
(API::SharedJSContext::ensureContext):
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView inspectable]):
(-[WKWebView setInspectable:]):
(-[WKWebView _allowsRemoteInspection]):
(-[WKWebView _setAllowsRemoteInspection:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
* Source/WebKit/UIProcess/Inspector/mac/WKInspectorViewController.mm:
(-[WKInspectorViewController webView]):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::m_fullscreenVideoTextRecognitionTimer):
(WebKit::WebPageProxy::inspectable const):
(WebKit::WebPageProxy::setInspectable):
(WebKit::WebPageProxy::allowsRemoteInspection const): Deleted.
(WebKit::WebPageProxy::setAllowsRemoteInspection): Deleted.
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/WebKit2Prefix.h:
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(-[WebView _commonInitializationWithFrameName:groupName:]):
(-[WebView initSimpleHTMLDocumentWithStyle:frame:preferences:groupName:]):
(-[WebView allowsRemoteInspection]):
(-[WebView setAllowsRemoteInspection:]):
* Tools/MiniBrowser/mac/WK2BrowserWindowController.m:
(-[WK2BrowserWindowController awakeFromNib]):
* Tools/MobileMiniBrowser/MobileMiniBrowserFramework/WebViewController.m:
(-[WebViewController createWebView]):
* Tools/WebKitTestRunner/cocoa/TestRunnerWKWebView.mm:
(-[TestRunnerWKWebView initWithFrame:configuration:]):

Canonical link: <a href="https://commits.webkit.org/255361@main">https://commits.webkit.org/255361@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/96a53ca470823468d91981a589ffdcde85f7b766

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/92238 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/1468 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/22824 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/102023 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/162292 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/96238 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/1465 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/29860 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/84678 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/98188 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/97895 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/956 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/78758 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/27907 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/82873 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/82536 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/70942 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/83651 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/36281 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/16495 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/78650 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/34037 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/17660 "Passed tests") | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/46/builds/27215 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3718 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/37910 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/40291 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/81272 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/39809 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/36801 "Passed tests") | [  ~~🧪 jsc-mips-tests~~](https://ews-build.webkit.org/#/builders/45/builds/18096 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
<!--EWS-Status-Bubble-End-->

dcrousso commented on Source/JavaScriptCore/API/JSContextPrivate.h:
```diff
@@ -69,7 +69,7 @@
 @discussion Remote inspection setting of the JSContext. Default value is YES.
 */
-@property (setter=_setRemoteInspectionEnabled:) BOOL _remoteInspectionEnabled JSC_API_AVAILABLE(macos(10.10), ios(8.0));
+@property (setter=_setRemoteInspectionEnabled:) BOOL _remoteInspectionEnabled JSC_API_DEPRECATED_WITH_REPLACEMENT("inspectable", macos(JSC_MAC_TBA, JSC_MAC_TBA), ios(JSC_IOS_TBA, JSC_IOS_TBA));
```
we should still have the old numbers
```
JSC_API_DEPRECATED_WITH_REPLACEMENT("inspectable", macos(10.10, JSC_MAC_TBA), ios(8.0, JSC_IOS_TBA))
```

dcrousso commented on Source/WTF/wtf/spi/ios/MobileGestaltSPI.h:
```diff
@@ -54,6 +54,7 @@ static const CFStringRef kMGQDeviceSupportsARKit CFSTR("arkit");
 static const CFStringRef kMGQTimeSyncCapability CFSTR("LJ8aZhTg8lXUeVxHzT+hMw");
 static const CFStringRef kMGQWAPICapability CFSTR("wapi");
 static const CFStringRef kMGQMainDisplayRotation CFSTR("MainDisplayRotation");
+static const CFStringRef kMGQCarrierInstallCapability CFSTR("CarrierInstallCapability");
```
hmmm some part of me finds it odd that we're doing this in WTF instead of JSC 🤔

dcrousso commented on Source/JavaScriptCore/API/JSContextRefPrivate.h:
```diff
@@ -101,15 +101,15 @@ JS_EXPORT void JSContextGroupClearExecutionTimeLimit(JSContextGroupRef group) JS
 @discussion Remote inspection is true by default.
 */
-JS_EXPORT bool JSGlobalContextGetRemoteInspectionEnabled(JSGlobalContextRef ctx) JSC_API_AVAILABLE(macos(10.10), ios(8.0));
+JS_EXPORT bool JSGlobalContextGetRemoteInspectionEnabled(JSGlobalContextRef ctx) JSC_API_AVAILABLE(macos(10.10), ios(8.0)) JSC_API_DEPRECATED_WITH_REPLACEMENT("JSGlobalContextIsInspectable", macos(10.10, JSC_MAC_TBA), ios(8.0, JSC_IOS_TBA));
```
I don't think we keep the `API_AVAILABLE` around when we add a `API_DEPRECATED_WITH_REPLACEMENT`

dcrousso commented on Source/JavaScriptCore/API/JSRemoteInspector.h:
```diff
@@ -47,8 +47,8 @@ JS_EXPORT void JSRemoteInspectorSetParentProcessInformation(JSProcessID, const u
 
-JS_EXPORT bool JSRemoteInspectorGetInspectionEnabledByDefault(void) JSC_API_AVAILABLE(macos(10.11), ios(9.0));
-JS_EXPORT void JSRemoteInspectorSetInspectionEnabledByDefault(bool) JSC_API_AVAILABLE(macos(10.11), ios(9.0));
+JS_EXPORT bool JSRemoteInspectorGetInspectionEnabledByDefault(void) JSC_API_AVAILABLE(macos(10.11), ios(9.0)) JSC_API_DEPRECATED("Use JSGlobalContextIsInspectable on a single JSGlobalContextRef.", macos(10.11, JSC_MAC_TBA), ios(9.0, JSC_IOS_TBA));
```
I don't think we keep the `API_AVAILABLE` around when we add a `API_DEPRECATED_WITH_REPLACEMENT`

dcrousso commented on Source/JavaScriptCore/API/JSRemoteInspector.cpp:
```diff
@@ -68,9 +77,61 @@ void JSRemoteInspectorSetLogToSystemConsole(bool logToSystemConsole)
     JSGlobalObjectConsoleClient::setLogToSystemConsole(logToSystemConsole);
 }
 
+static bool mainProcessHasEntitlement(WTF::ASCIILiteral entitlement, std::optional<audit_token_t> parentProcessAuditToken)
```
Should this function be inside a `#if PLATFORM(COCOA)`?  Or does `audit_token_t` exist elsewhere?

NIT: Is the `WTF::` necessary?

patrickangle replied:
Yes, this should be wrapped in `#if PLATFORM(COCOA)`, this is why it doesn't compile on WinCairo (and likely others) currently.

dcrousso commented on Source/JavaScriptCore/Configurations/JavaScriptCore.xcconfig:
```diff
@@ -29,14 +29,17 @@ MODULEMAP_FILE = $(SRCROOT)/JavaScriptCore.modulemap;
 WK_NO_STATIC_INITIALIZERS_Production_NO_ = $(WK_ERROR_WHEN_LINKING_WITH_STATIC_INITIALIZERS);
 
-OTHER_LDFLAGS_BASE = $(OTHER_LDFLAGS_HIDE_SYMBOLS) -force_load "$(WTF_ARCHIVE)" $(SOURCE_VERSION_LDFLAGS) $(WK_NO_STATIC_INITIALIZERS);
+OTHER_LDFLAGS_BASE = $(OTHER_LDFLAGS_HIDE_SYMBOLS) -force_load "$(WTF_ARCHIVE)" $(SOURCE_VERSION_LDFLAGS) $(WK_NO_STATIC_INITIALIZERS) $(WK_MOBILE_GESTALT_LDFLAGS);
```
please check with the JSC folks that this is OK

patrickangle replied:
Yusuke has potential RAM usage concerns, A/B testing underway to verify.

dcrousso commented on Source/JavaScriptCore/inspector/remote/RemoteInspectionTarget.cpp:
```diff
@@ -32,21 +32,37 @@
     m_allowed = allowed;
 
-    if (m_allowed && automaticInspectionAllowed())
+    if (remoteDebuggingAllowed() && automaticInspectionAllowed())
```
NIT: should we also change the `ASSERT(m_allowed)` below?

dcrousso commented on Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
```diff
@@ -202,7 +202,7 @@ for this property.
 - (_WKSessionState *)_sessionStateWithFilter:(BOOL (^)(WKBackForwardListItem *item))filter;
 
-@property (nonatomic, setter=_setAllowsRemoteInspection:) BOOL _allowsRemoteInspection;
+@property (nonatomic, setter=_setAllowsRemoteInspection:) BOOL _allowsRemoteInspection WK_API_DEPRECATED_WITH_REPLACEMENT("inspectable", macos(WK_MAC_TBA, WK_MAC_TBA), ios(WK_MAC_TBA, WK_IOS_TBA));
```
I think this should be `WK_API_DEPRECATED_WITH_REPLACEMENT("inspectable", macos(10.10, WK_MAC_TBA), ios(8.0, WK_IOS_TBA))`

dcrousso commented on Source/JavaScriptCore/API/JSContextRef.cpp:
```diff
@@ -249,6 +249,34 @@ void JSGlobalContextSetName(JSGlobalContextRef ctx, JSStringRef name)
+    VM& vm = globalObject->vm();
+    JSLockHolder lock(vm);
+
+    return globalObject->remoteDebuggingEnabled();
```
hmm why not also rename this to `inspectable`?  I thought the plan was to have `inspectable` control _all_ inspection, not just remote inspection?

dcrousso commented on Source/JavaScriptCore/API/JSRemoteInspector.cpp:
```diff
@@ -68,9 +73,55 @@ void JSRemoteInspectorSetLogToSystemConsole(bool logToSystemConsole)
+#if PLATFORM(MAC)
+        auto developerProvisioningEntitlement = "com.apple.security.get-task-allow"_s;
+#else
+        auto developerProvisioningEntitlement = "get_task_allow"_s;
```
underscores or dashes?  i thought it was `get-task-allow`?

dcrousso commented on Source/JavaScriptCore/inspector/remote/RemoteInspectionTarget.cpp:
```diff
@@ -32,21 +32,33 @@
 bool RemoteInspectionTarget::remoteControlAllowed() const
 {
-    return remoteDebuggingAllowed() || hasLocalDebugger();
+    return inspectable() || hasLocalDebugger();
```
ideally we'd entirely git rid of `hasLocalDebugger` and have everything just be off of the one `inspectable` flag

patrickangle replied:
I don't think we can entirely get rid of it because we need to inform the remote process that relays inspectable information that the process is being debugged in-process (RemoteInspectorCocoa.mm:497). But yeah we can probably get rid of its use here.

dcrousso replied:
to be clear this was more of a wishful thinking than something you have to change right now

i do think eventually we should move to a place where `inspectable` is the single source of truth for "can this be inspected" and then some new `inspected` would be the single source of truth for "is this actively being inspected"

dcrousso replied:
put another way, we should see if it's possible (eventually) to entirely remove the distinction between "local" vs "remote" inspection, instead only caring about whether or not something is inspected at all (not where it's coming from)

dcrousso commented on Source/JavaScriptCore/jsc.cpp:
```diff
@@ -3670,7 +3670,7 @@ void CommandLine::parseArguments(int argc, char** argv)
         }
 
         if (!strcmp(arg, "--remote-debug")) {
```
NIT: should we also rename this to `--inspect` or `--inspectable`? :P

dcrousso commented on Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
```diff
@@ -1427,6 +1427,23 @@ - (void)setInteractionState:(id)interactionState
+    return _page->inspectable();
+#else
+    return NO;
+#endif
```
wait so are we having `inspectable` be limited to remote inspection or not?  can we make this a full "line in the sand" moving away from remote inspection and instead be a global "can i be inspected", regardless of where the inspector is coming from?

patrickangle replied:
To an application using this framework/API on Cocoa platforms "remote inspection" is effectively equal to "inspection" because there is no supported API for in-process inspection. A global "inspectable" state is absolutely a long-term goal here, and the non-"remote" naming now will only help us later when we teach InspectorController how to use the "inspectable" property to determine if in-process inspection should be able to occur. That work is out of scope for this patch (but is an obvious "next step" when we have time).

So yes, we are trying to move away from the bifurcated idea in most places about remote v. in-process inspection, but this does not take us all the way there yet.

It is also worth noting that we will have to untangle at some point what is actually for remote inspection and what is for any inspection, since in most of JSC, remote inspection is the only inspection because there is no real way to do in-process inspection afaik, but JSC also has the shared inspection stuff that we need for both WebCore and JSC.

dcrousso replied:
ok if this is planned as a followup can we have a `// FIXME: ...`?

patrickangle replied:
Absolutely! I'll open a bugzilla to associate with the existing radar and try to hit key areas where the comment will be important.

dcrousso commented on Source/WebKit/UIProcess/Inspector/mac/WKInspectorViewController.mm:
```diff
@@ -95,6 +95,7 @@ - (WKWebView *)webView
     if (!_webView) {
         NSRect initialFrame = NSMakeRect(0, 0, WebKit::WebInspectorUIProxy::initialWindowWidth, WebKit::WebInspectorUIProxy::initialWindowHeight);
         _webView = adoptNS([[WKInspectorWKWebView alloc] initWithFrame:initialFrame configuration:self.webViewConfiguration]);
+        [_webView setInspectable:YES];
```
should we do this based on whether or not the "can inspect the inspector" setting is set?

patrickangle replied:
I think it is useful on macOS for a Web Inspector to be inspectable out-of-process even if the setting hasn't been toggled, since it might be in such a bad state that toggling the setting isn't practical.

dcrousso replied:
hmm i do worry that this might confuse developers tho if they happen to have both Safari and STP installed, as one might not realize that Web Inspector is a webpage

patrickangle replied:
Safari and STP today aren't able to inspect the other, and I expect that to remain true.

JosephPecoraro commented on Source/JavaScriptCore/API/JSContextRef.h:
```diff
@@ -143,19 +143,33 @@ JS_EXPORT JSGlobalContextRef JSContextGetGlobalContext(JSContextRef ctx) JSC_API
+/*!
+@function
+@abstract Gets whether the context is inspectable in Web Inspector.
+@param ctx The JSGlobalContext that you want to change the inspectability of.
```
Normally there is a `@result` for non-void functions.

patrickangle committed:
Remote Web Inspector: [Cocoa] `inspectable` API
https://bugs.webkit.org/show_bug.cgi?id=245986
rdar://100736163

Reviewed by Devin Rousso.

Add API to allow applications to explicitly mark a WKWebView or JSContext as inspectable. This flips assumptions about
inspectability of content. Previously the default state for inspectability was `true`, with a platform-specific deamon
making decisions about which applications would actually be inspectable on a system. This patch moves that decision into
WebKit/WebCore/JavaScriptCore instead, and flips the default for inspectability to be `false` unless an application
would have previously been always-inspectable. Those cases include:
- Applications with the `get-task-allow`/`com.apple.security.get-task-allow` development entitlements when linked
against an SDK without the API.
- Applications with the `com.apple.webinspector.allow`/`com.apple.private.webinspector.allow-remote-inspection`
entitlements.

These entitlements will no longer be necessary to enable inspection, and instead the new `inspectable` API should be
used.

* Source/JavaScriptCore/API/JSContext.h:
* Source/JavaScriptCore/API/JSContext.mm:
(-[JSContext inspectable]):
(-[JSContext setInspectable:]):
(-[JSContext _remoteInspectionEnabled]):
(-[JSContext _setRemoteInspectionEnabled:]):
* Source/JavaScriptCore/API/JSContextPrivate.h:
* Source/JavaScriptCore/API/JSContextRef.cpp:
(JSGlobalContextCreateInGroup):
(JSGlobalContextIsInspectable):
(JSGlobalContextSetInspectable):
(JSGlobalContextGetRemoteInspectionEnabled):
(JSGlobalContextSetRemoteInspectionEnabled):
* Source/JavaScriptCore/API/JSContextRef.h:
* Source/JavaScriptCore/API/JSContextRefPrivate.h:
* Source/JavaScriptCore/API/JSRemoteInspector.cpp:
(mainProcessHasEntitlement):
(defaultStateForRemoteInspectionEnabledByDefault):
(JSRemoteInspectorGetInspectionEnabledByDefault):
* Source/JavaScriptCore/API/JSRemoteInspector.h:
* Source/JavaScriptCore/API/tests/testIncludes.m:
* Source/JavaScriptCore/config.h:
* Source/JavaScriptCore/inspector/JSGlobalObjectInspectorController.cpp:
(Inspector::JSGlobalObjectInspectorController::developerExtrasEnabled const):
* Source/JavaScriptCore/inspector/remote/RemoteInspectionTarget.cpp:
(Inspector::RemoteInspectionTarget::remoteControlAllowed const):
(Inspector::RemoteInspectionTarget::inspectable const):
(Inspector::RemoteInspectionTarget::setInspectable):
(Inspector::RemoteInspectionTarget::pauseWaitingForAutomaticInspection):
(Inspector::RemoteInspectionTarget::setRemoteDebuggingAllowed): Deleted.
* Source/JavaScriptCore/inspector/remote/RemoteInspectionTarget.h:
* Source/JavaScriptCore/inspector/remote/RemoteInspector.h:
* Source/JavaScriptCore/inspector/remote/cocoa/RemoteInspectorCocoa.mm:
(Inspector::RemoteInspector::parentProcessAuditToken):
(Inspector::RemoteInspector::listingForInspectionTarget const):
* Source/JavaScriptCore/inspector/remote/glib/RemoteInspectorGlib.cpp:
(Inspector::RemoteInspector::listingForInspectionTarget const):
* Source/JavaScriptCore/inspector/remote/socket/RemoteInspectorSocket.cpp:
(Inspector::RemoteInspector::listingForInspectionTarget const):
* Source/JavaScriptCore/jsc.cpp:
(CommandLine::parseArguments):
(runJSC):
* Source/JavaScriptCore/runtime/JSGlobalObject.cpp:
(JSC::JSGlobalObject::setInspectable):
(JSC::JSGlobalObject::inspectable const):
(JSC::JSGlobalObject::setRemoteDebuggingEnabled): Deleted.
(JSC::JSGlobalObject::remoteDebuggingEnabled const): Deleted.
* Source/JavaScriptCore/runtime/JSGlobalObject.h:
* Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.h:
* Source/WebCore/config.h:
* Source/WebCore/page/Page.cpp:
(WebCore::Page::inspectable const):
(WebCore::Page::setInspectable):
(WebCore::Page::remoteInspectionAllowed const): Deleted.
(WebCore::Page::setRemoteInspectionAllowed): Deleted.
* Source/WebCore/page/Page.h:
* Source/WebCore/workers/service/context/ServiceWorkerThreadProxy.cpp:
(WebCore::ServiceWorkerThreadProxy::ServiceWorkerThreadProxy):
* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(WKPageGetAllowsRemoteInspection):
(WKPageSetAllowsRemoteInspection):
* Source/WebKit/UIProcess/API/Cocoa/APISerializedScriptValueCocoa.mm:
(API::SharedJSContext::ensureContext):
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView inspectable]):
(-[WKWebView setInspectable:]):
(-[WKWebView _allowsRemoteInspection]):
(-[WKWebView _setAllowsRemoteInspection:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
* Source/WebKit/UIProcess/Inspector/mac/WKInspectorViewController.mm:
(-[WKInspectorViewController webView]):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::m_fullscreenVideoTextRecognitionTimer):
(WebKit::WebPageProxy::inspectable const):
(WebKit::WebPageProxy::setInspectable):
(WebKit::WebPageProxy::allowsRemoteInspection const): Deleted.
(WebKit::WebPageProxy::setAllowsRemoteInspection): Deleted.
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/WebKit2Prefix.h:
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(-[WebView _commonInitializationWithFrameName:groupName:]):
(-[WebView initSimpleHTMLDocumentWithStyle:frame:preferences:groupName:]):
(-[WebView allowsRemoteInspection]):
(-[WebView setAllowsRemoteInspection:]):
* Tools/MiniBrowser/mac/WK2BrowserWindowController.m:
(-[WK2BrowserWindowController awakeFromNib]):
* Tools/MobileMiniBrowser/MobileMiniBrowserFramework/WebViewController.m:
(-[WebViewController createWebView]):
* Tools/WebKitTestRunner/cocoa/TestRunnerWKWebView.mm:
(-[TestRunnerWKWebView initWithFrame:configuration:]):

Canonical link: https://commits.webkit.org/255361@main
